### PR TITLE
DDS-1209 Add card component

### PR DESCRIPTION
**Pull request by rinjInTokyo**
This PR adds Card (`daikin-card`) and Card Title (`daikin-card-title`)  component.

## Related Ticket(s) and Links

- JIRA Ticket: [DDS-1209](https://daikinsvoil.atlassian.net/browse/DDS-1209)
- Figma design: [Figma](https://www.figma.com/design/VyaaU8Ta9yzyf0PsURWSSf/DDS%3A-Design-Kit?node-id=3406-2962&m=dev)

## API Design

- [JIRA Wiki](https://daikinsvoil.atlassian.net/wiki/spaces/GDS/pages/3103522843/Tile+Component+API)
- [JIRA Wiki](https://daikinsvoil.atlassian.net/wiki/spaces/GDS/pages/3103555618/DRAFT+Tile+Title+Component+API)

### Visual Regression Tests

- Card
    -  \[normal; withHeader;\] with header and \[none; gray; red\] border type.
- Card Title
    -  \[normal; withUnderline;\] underline type

### Interaction Tests

- Because it's a wrap component so not interaction now

## Checklist

- [x] I have added a changeset file that describes the changes.  
       _Hint_: run `npx changeset`.
- [x] I have performed a self-review of my code.
- [x] I have added appropriate visual regression tests.
- [x] I have added appropriate interaction tests.
- [x] I have confirmed that there are no accessibility warnings that can be addressed.  
       _Hint_: check the Accessibility tab in the Storybook.
- [x] I have confirmed that the component can be operated using only the keyboard.
- [x] I have confirmed that the focus state of the component is visually discernible.

[DDS-1209]: https://daikinsvoil.atlassian.net/browse/DDS-1209?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hiro-daikin** [comment]
I'm not convinced we need a specific component for `daikin-card-title`.
While I do like that having this component will enforce consistency, I think there are too many permutations of a `Card`, to where maybe some more flexibility is needed.

Brad Frost thinks of Cards as multiple components which composes [recipes](https://bradfrost.com/blog/post/the-art-of-design-system-recipes/). 
The article also has this [example](https://codepen.io/bradfrost/pen/RwOMeap) which may be easier to follow

I like shoelace's [implementation](https://shoelace.style/components/card#slots), where they have `header` and `footer` slots. The user can specify the content of the slots. 

Please let me know of your thoughts.
cc: @yodas7

File: src/components/card-title/daikin-card-title.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+ * Hierarchy:
+ * - `daikin-card` > `daikin-card-title`
+ *
+ * @slot icon - A slot for card title icon. Place `daikin-icon` element here.
+ *
+ * @slot label - A slot for card title Text content. Place a text element here.
+ *
+ * @slot link - A slot for card title link. Place link elements here.
+ *
+ * @slot action - A slot for card title button. Place `daikin-button` elements here.
```
Comment (yodas7) [inline src/components/card-title/daikin-card-title.ts:52]:
```suggestion
 * @slot icon - A slot for card title icon. Place a `daikin-icon` element here.
 * @slot label - A slot for card title text content. Place a text here.
 * @slot link - A slot for card title link. Place a link element here.
 * @slot action - A slot for card title button. Place a `daikin-button` element here.
```

File: src/components/card-title/daikin-card-title.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import { customElement, property } from "lit/decorators.js";
+
+import { cva } from "class-variance-authority";
+import tailwindStyles from "../../tailwind.css?inline";
+
+const cvaContainer = cva(
+  ["flex", "items-center", "w-full", "justify-between", "last:pr-6"],
+  {
+    variants: {
+      underLine: {
```
Comment (yodas7) [inline src/components/card-title/daikin-card-title.ts:11]:
```suggestion
      underline: {
```

File: src/components/card-title/daikin-card-title.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      display: flex;
+      align-items: center;
+    }
+  `;
+
+  /**
+   * If `true` will show a blue line under the card
+   */
+  @property({ type: Boolean, reflect: true, attribute: "under-line" })
+  underLine = false;
```
Comment (yodas7) [inline src/components/card-title/daikin-card-title.ts:81]:
```suggestion
  @property({ type: Boolean, reflect: true })
  underline = false;
```

File: src/components/card-title/daikin-card-title.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  override render() {
+    const containerClassName = cvaContainer({
+      underLine: this.underLine,
+    });
+
+    const labelClassName = cvaLabel({
+      withIcon: this.withIcon,
+    });
+
+    return html` <div class="${containerClassName}">
```
Comment (yodas7) [inline src/components/card-title/daikin-card-title.ts:94]: nit.

```suggestion
    return html` <div class=${containerClassName}>
```

**Review by yodas7 — COMMENTED** [review]
Made some comments.

BTW what is the status of this component?
If the design has not been finalized, please add a `waiting for design` label and keep it as a draft.

**Comment by yodas7** [comment]
@hiro-daikin 
I think it depends on the design philosophy of the UI library.
There are libraries that provide components for card titles, such as MUI and Chakra UI, but there are also libraries that don't.

So far, we have prepared all the components exist in Figma, so if we follow that, we will have to create them.
However, it is debatable whether that is the best way to go forward. In other words, as you said, we could decide to prepare components up to a certain extent, but leave the rest to the user.
I think that this would require a decision not only by the engineers, but also by the team as a whole.

**Comment by hiro-daikin** [comment]
@yodas7 
I think you are right that we need to be clear as a team whether or not we want to be very specific or give users more flexibility. 

The background for comment is also the fact that we may want to add a `text` component:
- h1
- h2
- h3
- regular
- etc

If we decide to be very strict, instead of the `daikin-card-title`, we can also reuse this text component. 
If we want to be flexible, we can expose a `slot` and users can specify which text variant they want to use.

**Comment by yodas7** [comment]
It has been decided that the design will be revised. lin-san will provide feedback to the designer.

**Pull request closed without merging**